src/cli/build.rs

```diff
@@ -21,53 +40,243 @@ pub struct BuildCommand {
     /// input file path with an executable extension
     #[clap(short, long)]
     pub output: Option<PathBuf>,
+
+    /// The target to compile for - defaults to the host triple
+    #[clap(short, long)]
+    pub target: Option<String>,
+
+    /// The path to the base executable to use - defaults to
+    /// the currently running executable, used for cross-compilation
+    /// for targets not directly supported
+    #[clap(short, long)]
+    pub base: Option<PathBuf>,
```

Comment:
I understand why you implemented it, but I'm unsure if supporting it is actually a good idea. It puts the burden on Lune to document the option and all its caveats, and provide support for it. Advanced users could (and probably should) use `cargo run --release -- build ...` or similar from their wanted platform for compiling unsupported binaries instead.

Reply:
Releasing precompiled binaries for many platforms is a greater burden (considering there are many different platforms we cannot may not have access to - for example, windows devices, which are already a bit hard to test in practice), than allowing a user to provide a binary themselves, and warning them that it is at their risk, and not officially supported by lune. We could print a warning message for the flag, maybe.

Reply:
It is a greater burden to officially support something, at all. I'm suggesting that ***if*** we do so, we do it by supporting it fully instead of supporting a single step in the process.

1. People still need to get their nonstandard Lune binary from somewhere. We don't provide any outside of releases.
2. At some point, you need to build Lune from source to then build a custom standalone binary.
3. Telling people to use something like `cargo run --release -- build` if they're curious about building their own binaries is a single step process. Telling them to compile, find the compiled binary, move it somewhere known, add an extra argument to `build`, is not.

Leveraging cargo for custom builds here already works great, will work after this is merged, and will work for the foreseeable future without us having to support anything new.

Reply:
Removed in https://github.com/lune-org/lune/pull/162/commits/971eccf12bb272f2d212a2e7dcf171937d83af2f.